efi/uc_uki_load_handler.go

```diff
@@ -0,0 +1,39 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package efi
+
+import (
+	"errors"
+)
+
+type ubuntuCoreUKILoadHandler struct{}
+
+func newUbuntuCoreUKILoadHandler(_ peImageHandle) (imageLoadHandler, error) {
+	return new(ubuntuCoreUKILoadHandler), nil
+}
+
+func (h *ubuntuCoreUKILoadHandler) MeasureImageStart(_ pcrBranchContext) error {
+	// TODO: Add stuff that the kernel measures here
```

Comment:
do we need to fix this before things can be used?

Reply:
It doesn't - snapd can continue to use the `AddSystemdStubProfile` API in addition to this one, although that will be moved here in a follow up. That will be a fairly minor change though.

Reply:
maybe it's a good idea to leave a comment also about that:

// for now clients can continue using AddSystemdStubProfile